### PR TITLE
Fix #2432 Added versioning for main js and theme

### LINKED ISSUE #2432: Introduce mechanism to version JavaScript and other resources as needed
It is important that we introduce a versioning mechanism for api so that when we deploy a fix even in development we do not have to refresh the cache in the client everytime there is a fix.

### PR DESCRIPTION
**Pull request by MV88**
## Description
Provided a mechanism to version the mapstore.js file and the default.css 

## Issues
 - Fix #2432 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
Actually mapstore loads mapstore.js and default.css

**What is the new behavior?**
it loads them by attaching ?parameter where parameter changes on every build

**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] No

### CODE REVIEW

**Comment by MV88** [comment]
An update is being made for C028 project and this is related on when to render mapstore.
One way is to load it when the css of the theme is loaded. 

@mbarto did you want this to be optional/configurable?

**Comment by mbarto** [comment]
if possible

File: buildConfig.js [hunk 1]
```diff
@@ -129,15 +135,18 @@ module.exports = (bundles, themeEntries, paths, extractThemesPlugin, prod, publi
...
+        }] : [])
     },
     devServer: {
         proxy: {
             '/rest/geostore': {
                 target: "http://dev.mapstore2.geo-solutions.it/mapstore"
             },
-            '/proxy': {
-                target: "http://dev.mapstore2.geo-solutions.it/mapstore"
+            '/mapstore/proxy': {
```
Comment (mbarto) [inline buildConfig.js]: Revert this

File: web/pom.xml [hunk 2]
```diff
@@ -80,6 +80,29 @@
...
+                    <configuration>
+                        <outputDirectory>${basedir}/target/mapstore</outputDirectory>
+                        <encoding>UTF-8</encoding>
+                        <resources>
+                            <resource>
+                                <directory>${basedir}/client/dist</directory>
+                                <includes>
+                                    <include>index.html</include>
+                                </includes>
+                                <excludes>
```
[the remark this answers is not in the document]
Comment (MV88) [reply web/pom.xml]: quote from your resource 
> Of course, we can also have both <includes> and <excludes> elements
this execution section takes only "index.html".

what did you mean exactly with your comment?

**Comment by MV88** [comment]
Changes needed to complete this task:
[ ] make rendering of mapstore after theme has loaded optional & configurable
[ ] fix conflicts and buildConfig.js

File: web/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/pom.xml]: exclude seems not needed, see the doc to confirm

**Pull request merged**